library/Notifications/Hook/ObjectsRendererHook.php

```diff
@@ -61,30 +64,58 @@ abstract public function getSourceType(): string;
     abstract public function createObjectLink(array $objectIdTag): ValidHtml;
 
     /**
-     * Register HTMLs to be rendered for the object names using the cached object ID tags
+     * Register object ID tags to the cache
+     *
+     * @param Objects $obj
      *
      * @return void
      */
-    final public static function registerHtmlForObjectNames(): void
+    final public static function register(Objects $obj): void
+    {
+        self::$objectIdTags[$obj->source->type][$obj->id] = $obj->id_tags;
+    }
+
+    /**
+     * Load HTMLs to be rendered for the object names to the cache using the cached objects
+     *
+     * @return void
+     */
+    final public static function load(): void
+    {
+        self::prepare(self::$objectIdTags);
+    }
+
+    /**
+     * Prepare the objects to be rendered using the given object ID tags for each source
+     *
+     * The supplied object ID tags must have the following structure:
+     * ['object source type' => ['object ID' => [object ID tags]]].
+     *
+     * @param array<string, array<string, array<string, string>>> $objectIdTags Array of object ID tags for each source
+     *
+     * @return void
+     */
+    private static function prepare(array $objectIdTags): void
     {
         $objectDisplayNames = [];
         foreach (Hook::all('Notifications\\ObjectsRenderer') as $hook) {
             /** @var self $hook */
             try {
-                $objectDisplayNames[] = $hook->getHtmlForObjectNames(self::$objectIdTagsCache[$hook->getSourceType()]);
+                $objectDisplayNames[] = $hook->getHtmlForObjectNames($objectIdTags[$hook->getSourceType()]);
+
+                unset($objectDisplayNames[$hook->getSourceType()]);
```

Comment:
I would prefer to reset the entire array after the loop. If there are 5 hooks, we call unset 5 times, which is unnecessary in my opinion.